refinery/ui/source/js/tool-launch/ctrls/input-control-inner-nav-ctrl.js

```diff
@@ -30,36 +32,10 @@
    * Methods Definitions
    * ---------------------------------------------------------
    */
-    // helper method to check if each group-inputType are empty
-    function areInputFileTypesEmpty () {
-      var isEmpty = false;
-      angular.forEach(fileService.groupCollection[vm.currentGroup], function (inputList) {
-        if (inputList.length === 0) {
-          isEmpty = true;
-        }
-      });
-      return isEmpty;
-    }
 
     // View method to check if the group has minimum nodes
     function needMoreNodes () {
-      var moreFlag = false;
-      var groupType = fileService.currentTypes[fileService.currentTypes.length - 1];
-      if (!_.property(vm.currentGroup)(fileService.groupCollection)) {
-        moreFlag = true;
-      } else if (groupType === 'PAIR') {
-        // pair, requires 2 inputTypes
-        var inputLength = _.keys(fileService.groupCollection[vm.currentGroup]).length;
-        if (inputLength > 1) {
-          moreFlag = areInputFileTypesEmpty();
-        } else {
-          moreFlag = true;
-        }
-      } else {
-        // list
-        moreFlag = areInputFileTypesEmpty();
-      }
-      return moreFlag;
+      return toolLaunchService.checkNeedMoreNodes();
```

Comment:
I'd leave the function name the same. I think it's more natural to read/write `if needMoreNodes()`.